atom/browser/api/atom_api_web_contents.h

```diff
@@ -133,6 +133,10 @@ class WebContents : public mate::TrackableObject<WebContents>,
   void SetAllowTransparency(bool allow);
   bool IsGuest() const;
 
+  // Handler for permission requests.
+  void SetPermissionRequestHandler(v8::Local<v8::Value> val,
```

Comment:
if we are providing this method with `session` module, can the `permission-request` event be removed ? the event way currently breaks permission being granted by default. With the callback,

```
session.setPermissionRequestHandler((webContents, permission, callback) => {
  // logic
  callback(); // "granted/denied"
})
```